jenkins-master/bin/setup-and-run-jenkins.sh

```diff
@@ -4,17 +4,50 @@
 # CA certificate to it. This allows Jenkins to use the certificate when
 # connecting to Kubernetes API.
 
-CONFIG_PATH="/var/lib/jenkins/config.xml"
-KUBE_CA="/run/secrets/kubernetes.io/serviceaccount/ca.crt"
-STORE_PATH="/var/lib/jenkins/keystore"
+CONFIG_PATH=/var/lib/jenkins/config.xml
+
+PROJECT_NAME=${PROJECT_NAME:-ci}
+OPENSHIFT_API_URL=https://openshift.default.svc.cluster.local
+KUBE_CA=/run/secrets/kubernetes.io/serviceaccount/ca.crt
+AUTH_TOKEN=/run/secrets/kubernetes.io/serviceaccount/token
+K8S_PLUGIN_POD_TEMPLATES=""
+
+# Here we filter all image streams with label set to 'role=jenkins-slave' and
+# use their dockerImageRepository to generate slave definitions that Kubernetes
+# plugin will use.
+#
+# Each imageStream must have two labels set:
+# * 'role=jenkins-slave'  - To mark this imageStream as Jenkins Slave compatible
+# * 'jenkins-label=LABEL' - To set the Jenkins 'label' for the slave. You have
+#                           to use this label in your Jenkins job definition
+#
+alias oc="oc -n ${PROJECT_NAME} --token=$AUTH_TOKEN --server=$OPENSHIFT_API_URL --certificate-authority=${KUBE_CA}"
+IFS='|' read -ra slave_images <<< $(oc get is -l role=jenkins-slave -o template --template="{{range .items}}{{index .metadata.labels \"jenkins-label\"}};{{.status.dockerImageRepository}}|{{end}}")
```

Comment:
@gabemontero @bparees FYI

This will allow Jenkins to bootstrap Kubernetes Slaves from your existing ImageStreams that has "jenkins-slave"  and "jenkins-label" set.